juju/juju
# [JUJU-447] Process offer permissions when migrating models

When migrating a model, the offer permissions we not being copied across.
The PR fixes that, and also ensures the permissions are removed from the source model.

## QA steps

```
juju bootstrap lxd c1
juju add-model foo
juju deploy mariadb && juju offer mariadb:db

juju bootstrap lxd c2
juju switch c1
juju migrate foo c2
juju switch c2

juju show-offer mariadb
```

(check that there are 2 permissions)
migrate model back to c1
```
juju migrate foo c1
juju switch c2
juju show-offer mariadb
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1957745
